_posts/2024-12-18-cockpit-331.md

```diff
@@ -0,0 +1,71 @@
+---
+title: Cockpit 331
+author: jelle
+date: '2024-12-18'
+tags: cockpit, machines, files
+slug: cockpit-331
+category: release
+summary: 'Prevent search engine indexing, support sharing host ssh-agent, downloading of unsupported operating systems and file upload improvements'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly.
+
+Here are the release notes from Cockpit 331, cockpit-machines 326, and cockpit-files 14:
+
+
+## ws: Prevent search engine indexing with robots.txt
+
+Public Cockpit instances are no longer indexed by default.
+
+## ws container: Support sharing host ssh-agent
+
+For SSH key authentication, the [cockpit/ws container](https://quay.io/repository/cockpit/ws) has supported bind-mounting private SSH keys into the container for a long time. That mode is appropriate for server system containers or deploying in e.g. Kubernetes.
+
+For desktop use cases similar to [Cockpit Client](https://github.com/flathub/org.cockpit_project.CockpitClient) it is preferable to instead run the ws container as your own user, and share your user session's [SSH agent](https://linux.die.net/man/1/ssh-agent). This provides a more comfortable login experience as you don't have to unlock private keys with your passphrases on the Cockpit login page again, and this also avoids exposing the private key to the web server.
+
+Please see the "SSH authentication: Share SSH agent with container" section in the [container documentation](https://quay.io/repository/cockpit/ws) for details.
+
+## Machines: Download and install unsupported and older operating systems
+
+Cockpit now offers to download and install also unsupported and older operating systems, while still promoting supported and newer ones.
+
+![screenshot of download and install unsupported and older operating systems]({{ site.baseurl }}/images/331-machines-download-and-install-unsupported-and-older-operating-systems.png)
+
+## Files: Allow editing of multiple file permissions
+
+The permissions of multiple selected files can now be changed if they have the same owner and group.
+
+![screenshot of ## allow editing of multiple file permissions]({{ site.baseurl }}/images/331-allow-editing-of-multiple-file-permissions.png)
+
+## Files: Allow uploading files as administrator
+
+When logged in with administrator privileges files are now allowed to be uploaded, the default ownership of these files are the user/group of the current directory. The permissions of uploaded file(s) can be changed after the upload completes.
```

Comment:
Please verify that this is actually correct:

```suggestion
When logged in with administrator privileges, you can now upload files to directories other than your home. The default ownership of these files are the user/group of the current directory. This can be changed after the upload completes.
```

Reply:
Much better! Thanks!